src/networking.c

```diff
@@ -2596,6 +2607,57 @@ int processInputBuffer(client *c) {
     return C_OK;
 }
 
+/* This handler Buffer incoming steady-state replication data while downloading and loading RDB.
+ * The buffer will store the replication data until we finish loading the RDB, then we will stream
+ * it into the db, and continue using sds type query buffer */
+void bufferReplData(connection *conn) {
+    client *c = connGetPrivateData(conn);
+    int nread;
+    long long readlen = PROTO_IOBUF_LEN;
+    replDataBuf *repl_buf = server.repl_data_buf;
+
+    /* Update total number of reads on server */
+    atomicIncr(server.stat_total_reads_processed, 1);
+
+    /* Check if the replication buffer needs more space */
+    if (repl_buf->size - repl_buf->used < readlen) {
+        long long new_size = max(repl_buf->size * REPLICA_BUFFER_GROWTH_FACTOR, repl_buf->size + readlen);
+        if (server.replica_full_sync_buffer_limit > 0 && new_size > server.replica_full_sync_buffer_limit) {
+            new_size = server.replica_full_sync_buffer_limit;
+        }
+        if (repl_buf->size >= new_size) {
+            serverLog(LL_WARNING, "Replica's replication buffer limit reached %lld. Aborting rdb-chanel-sync.", repl_buf->size);
+            abortRdbConnectionSync(1);
+            return;
+        }
+        serverLog(LL_DEBUG, "Resize replication buffer to %lld.", new_size);
+        resizeReplicationBuffer(new_size);
+    }
+
+    nread = connRead(c->conn, repl_buf->buf + repl_buf->used, readlen);
+    if (nread == -1) {
+        if (connGetState(conn) == CONN_STATE_CONNECTED) {
+            return;
+        } else {
+            serverLog(LL_VERBOSE, "Reading from primary: %s",connGetLastError(c->conn));
+            abortRdbConnectionSync(1);
+            return;
+        }
+    } else if (nread == 0) {
+        if (server.verbosity <= LL_VERBOSE) {
+            sds info = catClientInfoString(sdsempty(), c);
+            serverLog(LL_VERBOSE, "Client closed connection %s", info);
+            sdsfree(info);
+        }
+        abortRdbConnectionSync(1);
+        return;
+    }
```

Comment:
```suggestion
    if (nread == -1) {
      if (connGetState(conn) != CONN_STATE_CONNECTED) {
          serverLog(LL_VERBOSE, "Reading from primary: %s", connGetLastError(c->conn));
          abortRdbConnectionSync(1);
      }
      return;
  }
  
  if (nread == 0) {
      if (server.verbosity <= LL_VERBOSE) {
          sds info = catClientInfoString(sdsempty(), c);
          serverLog(LL_VERBOSE, "Client closed connection %s", info);
          sdsfree(info);
      }
      abortRdbConnectionSync(1);
      return;
  }
```
If we return inside an IF Block, there is no need to use Else if. We can simply use If statement.

Reply:
Thanks 👍